.github/workflows/knative-downstream.yaml

```diff
@@ -72,15 +72,33 @@ jobs:
          git fetch upstream ${{ github.base_ref }}
          git pull --no-rebase --no-commit upstream ${{ github.base_ref }}
 
-    - name: Checkout downstream at ${{ github.base_ref }}
+    - name: Attempt checkout of downstream at ${{ github.base_ref }}
+      id: checkout-downstream
       uses: actions/checkout@v4
       with:
         ref: ${{ github.base_ref }}
         repository: ${{ matrix.repo }}
         path: downstream
+      continue-on-error: true
+
+    - name: Set branch to main if checkout fails
+      if: steps.checkout-downstream.outcome == 'failure'
+      run: echo "BRANCH=main" >> $GITHUB_ENV
+
+    - name: Set branch to PR base ref if checkout succeeds
+      if: steps.checkout-downstream.outcome == 'success'
+      run: echo "BRANCH=${{ github.base_ref }}" >> $GITHUB_ENV
+
+    - name: Checkout downstream with fallback branch
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ env.BRANCH }}
+        repository: ${{ matrix.repo }}
+        path: downstream
```

Comment:
We can skip this step if the earlier checkout succeeded

Reply:
Might be simpler to not use the env var and collapse the steps into something like this

```yaml
    - name: Checkout downstream with fallback branch
      if: steps.checkout-downstream.outcome == 'failure'
      uses: actions/checkout@v4
      with:
        ref: main
        repository: ${{ matrix.repo }}
        path: downstream
```

Reply:
Simplified steps as changes suggested